### PR TITLE
Add password reset

### PR DESCRIPTION
**Pull request by zpnk**
Two known "issues":
- The "/password_reset" patch endpoint can't necessarily be reused for a logged-in user wanting to change passwords. Perhaps this endpoint would be better served under a potential userApi: PATCH "/users/:id/password".  We could switch over when we eventually add a userApi.
- Existing user sessions aren't reset. We would need a way to find specific sessions in the db by `userId`. This functionality would be quite possible to add, but if we're planning on moving back to JWTs for auth it's not really worth doing atm. Once we revert, we can revisit this. (Hence the TODO :D)

PS: plzdontsquash :p

### CODE REVIEW

**Comment by zpnk** [comment]
Rebased.

File: src/server/authApi.js [hunk 1]
```diff
@@ -85,4 +90,58 @@ router.delete('/session', async (request, response) => {
...
+  const resetToken = jwt.sign({userId: user.id}, AUTH_SECRET, {
+    jwtid,
+    expiresIn: '10m'
+  });
+
+  await user.update({resetPasswordToken: jwtid});
+
+  const resetUrl = `${request.protocol}://${request.get('Host')}/reset?token=${resetToken}`;
+
+  if (NODE_ENV === 'production') {
```
Comment (amccloud) [inline src/server/authApi.js]: Can we remove this condition? Later on we can create a mock SMTP server that logs to the console in development.

File: src/data/models/user.js [hunk 2]
```diff
@@ -3,7 +3,8 @@ module.exports = (sequelize, DataTypes) => (
     email: DataTypes.STRING,
     password: DataTypes.STRING,
     displayName: DataTypes.STRING,
-    avatarUrl: DataTypes.STRING
+    avatarUrl: DataTypes.STRING,
+    resetPasswordToken: DataTypes.STRING
```
Comment (amccloud) [inline src/data/models/user.js]: NIT: Can we name this `resetPasswordTokenId` to clarify the difference between the token and the id used to generate the token?

**Comment by zpnk** [comment]
Fixed!

**Pull request merged**